File: BiblioTec/DTOs/PrestamoDto.cs
namespace BiblioTec.DTOs.Prestamos
{
    public class PrestamoDto
    {
        public int PrestamoId { get; set; }

        public int UsuarioId { get; set; }
        public string NombreUsuario { get; set; }

        public int LibroId { get; set; }
        public string TituloLibro { get; set; }
        public string AutorLibro { get; set; }

        public DateTime FechaPrestamo { get; set; }
        public DateTime FechaVencimiento { get; set; }
        public DateTime? FechaDevolucion { get; set; }

        public string Estado { get; set; }
    }

    
    public class PrestamoCreateDto
    {
        public int LibroId { get; set; }
    }

    public class PrestamoResumenDto
    {
        public int PrestamoId { get; set; }
        public int LibroId { get; set; }
        public string TituloLibro { get; set; }
        public string AutorLibro { get; set; }
        public string RutaPdf { get; set; }
        public DateTime FechaPrestamo { get; set; }
        public DateTime FechaVencimiento { get; set; }
        public string Estado { get; set; }
    }
}

File: BiblioTecAdmin/src/main/java/com/bibliotec/admin/model/PrestamoDto.java
package com.bibliotec.admin.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@JsonIgnoreProperties(ignoreUnknown = true)
public class PrestamoDto {
    private static final DateTimeFormatter FMT = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");

    public int idPrestamo;
    public int idUsuario;
    public String nombreUsuario;
    public int idLibro;
    public String tituloLibro;
    public String autorLibro;
    public LocalDateTime fechaPrestamo;
    public LocalDateTime fechaVencimiento;
    public LocalDateTime fechaDevolucion;
    public String estado;

    public int    getIdPrestamo()      { return idPrestamo; }
    public int    getIdUsuario()       { return idUsuario; }
    public String getNombreUsuario()   { return nombreUsuario != null ? nombreUsuario : ""; }
    public int    getIdLibro()         { return idLibro; }
    public String getTituloLibro()     { return tituloLibro != null ? tituloLibro : ""; }
    public String getAutorLibro()      { return autorLibro != null ? autorLibro : ""; }
    public String getEstado()          { return estado != null ? estado : ""; }
    public String getFechaPrestamoStr()    { return fechaPrestamo   != null ? fechaPrestamo.format(FMT)   : ""; }
    public String getFechaVencimientoStr() { return fechaVencimiento != null ? fechaVencimiento.format(FMT) : ""; }
    public String getFechaDevolucionStr()  { return fechaDevolucion  != null ? fechaDevolucion.format(FMT)  : "Pendiente"; }
}
